django/db/backends/sqlite3/base.py

```diff
@@ -399,13 +404,17 @@ def _sqlite_datetime_parse(dt, tzname):
 
 def _sqlite_datetime_cast_date(dt, tzname):
     dt = _sqlite_datetime_parse(dt, tzname)
+    if type(dt) is datetime.date:
```

Comment:
Is this addition and the new `isinstance` check below tested? I removed them and I don't see any failing tests.